Refactor logging package to split syslog functionality into separate file

tklauser opened the pull request:
This will allow to build `pkg/logging` on platforms not supporting syslog for `cilium-cli`, see https://github.com/cilium/cilium-cli/issues/231. Thus also marking for backports to 1.10 since we vendor a stable `cilium/cilium` release into `cilium-cli` (currently `v1.9.8`, but eventually switching to `v1.10`).

See individual commits for details.

rolinh commented on pkg/logging/logging_nosyslog.go:
```diff
@@ -0,0 +1,21 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !linux
```
```suggestion
// +build windows
```

rolinh commented on pkg/logging/logging_syslog.go:
```diff
@@ -0,0 +1,165 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
```
```suggestion
// +build !windows
```

rolinh reviewed (changes requested):
Thanks, LGTM with one change requested. As syslog is not Linux specific, I'd rather invert the build tags to explicitly exclude syslog from Windows build.

tklauser committed:
logging: factor complete syslog setup into setupSyslog

Move all syslog setup steps into function setupSyslog. No functional
changes, this is in preparation for moving syslog specific functionality
to a build tag protected file for non-linux build support.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
logging: split syslog functionality into separate file

This package is a transitive dependency in cilium-cli, but currently it
fails to build on platforms without syslog support (e.g. windows).
However, we want still to be able to build it for some these platforms,
see e.g. https://github.com/cilium/cilium-cli/issues/231. Fix the build
by moving all functionality specifiy to syslog into a separate file,
protected by build tags. The functionality will be stubbed out on
platforms without syslog.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
